apps/common-app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/UpdatesContainer.ts

```diff
@@ -87,70 +84,110 @@ export function createUpdatesContainer(testRunner: TestRunner) {
+    if ('backgroundColor' in updatesCopy) {
+      updatesCopy.backgroundColor = convertDecimalColor(updatesCopy.backgroundColor);
+    }
+    _updateNativeSnapshot([{ tag, update }], jsUpdates.value.length - 1);
     jsUpdates.modify(updates => {
       updates.push({
         tag,
-        // Deep Copy, works with nested objects, but doesn't copy functions (which should be fine here)
-        update: JSON.parse(JSON.stringify(update)),
+        update: updatesCopy,
       });
       return updates;
     });
   }
 
-  function getUpdates(propsNames: string[] = []) {
-    const updates: OperationUpdate[] = [];
-    if (propsNames.length === 0) {
-      for (const updateRequest of jsUpdates.value) {
-        updates.push(updateRequest.update);
+  function _sortUpdatesByViewTag(
+    updates: Array<JsUpdate> | Array<NativeUpdate>,
+    propsNames: string[],
+  ): [MultiViewSnapshot, 'multiView'] | [SingleViewSnapshot, 'singleView'] {
+    const updatesForTag: Record<number, Array<OperationUpdate>> = {};
+    for (const updateRequest of updates) {
+      const { tag } = updateRequest;
+
+      if (!(tag in updatesForTag)) {
+        updatesForTag[tag] = [];
       }
-    } else {
-      for (const updateRequest of jsUpdates.value) {
-        const filteredUpdate: Record<string, OperationUpdate> = {};
+      let update: OperationUpdate = [];
+      if (propsNames.length === 0) {
+        update = 'update' in updateRequest ? updateRequest.update : updateRequest.snapshot;
+      } else {
         for (const prop of propsNames) {
-          filteredUpdate[prop] = updateRequest.update[prop as keyof OperationUpdate];
+          update[prop] =
+            'update' in updateRequest
+              ? updateRequest.update[prop as keyof OperationUpdate]
+              : updateRequest.snapshot[prop as keyof OperationUpdate];
         }
-        updates.push(filteredUpdate);
       }
+      updatesForTag[tag].push(update);
+    }
+
+    const recordedMultipleViews = Object.keys(updatesForTag).length > 1;
+
+    if (recordedMultipleViews) {
+      return [updatesForTag, 'multiView'];
+    } else {
+      // In case of recording only one view return an array
+      const singleViewSnapshot: SingleViewSnapshot = updatesForTag[Number(Object.keys(updatesForTag)[0])];
+      return [singleViewSnapshot, 'singleView'];
+    }
+  }
+
+  function getUpdates(component?: TestComponent, propsNames: string[] = []): SingleViewSnapshot {
+    const [sortedUpdates, snapshotType] = _sortUpdatesByViewTag(jsUpdates.value, propsNames);
+    if (snapshotType === 'multiView') {
+      if (component === undefined) {
+        throw new Error('Recorded snapshots of many views, specify component you want to get snapshot of');
+      }
+      const tag = component?.getTag();
+      if (!tag || !(tag in sortedUpdates)) {
+        throw new Error('Snapshot of given component not found');
+      } else {
+        return sortedUpdates[tag];
+      }
+    } else {
+      return sortedUpdates;
```

Comment:
I believe we can simplify this logic by always returning `return sortedUpdates[tag];`. The `_sortUpdatesByViewTag` function can consistently sort by tag. If there is only one view, you can return a dictionary with a single key. This approach would make the overall logic of this method simpler.